config/cmake/HDFCompilerFlags.cmake

```diff
@@ -56,6 +56,44 @@ if (CMAKE_C_COMPILER_ID STREQUAL "NVHPC" )
   else ()
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Mbounds -g")
   endif ()
+
+  # With at least NVHPC 23.5 - 23.9, compiling with -O2 or higher and -DNDEBUG
+  # appears to have issues that manifest in the tests as incorrect metadata
+  # checksums being read or memory being corrupted. Compiling without -DNDEBUG
+  # does not appear to have these issues, but is not ideal due to compiling in
+  # asserts and other library debug code. Compiling with -O1 also does not appear
+  # to have these issues, so set maximum optimization level to -O1 for now until
+  # it can be determined whether these issues are compiler-specific or issues
+  # in the library.
```

Comment:
Do we have a keyword that we can add to this to indicate we should revisit the issue at a later date?

Reply:
I thought about creating an issue to track this, but open to other suggestions as well.

Reply:
@derobins , we probably need to develop a procedure indicating workarounds that should be revisited at a future date. Otherwise, they will be forgotten about.